src/item_contents.cpp

```diff
@@ -1124,11 +1134,18 @@ std::list<item *> item_contents::all_items_top( const std::function<bool( item_p
     return all_items_internal;
 }
 
-std::list<item *> item_contents::all_items_top( item_pocket::pocket_type pk_type )
+std::list<item *> item_contents::all_items_top( item_pocket::pocket_type pk_type,
+        bool unloading )
 {
-    return all_items_top( [pk_type]( item_pocket & pocket ) {
-        return pocket.is_type( pk_type );
-    } );
+    if( unloading ) {
+        return all_items_top( [pk_type]( item_pocket & pocket ) {
+            return pocket.is_type( pk_type ) && pocket.settings.is_unloadable();
+        } );
+    } else {
+        return all_items_top( [pk_type]( item_pocket & pocket ) {
+            return pocket.is_type( pk_type );
+        } );
+    }
```

Comment:
```suggestion
    return all_items_top( [pk_type]( item_pocket & pocket ) {
          return pocket.is_type( pk_type ) && ( !unloading || pocket.settings.is_unloadable() );
    } );
```

Reply:
This is much harder to read and understand and a fraction slower I don't really think it's worth changing.

Reply:
Enough people disagreed that there's even a clang-tidy check for [this](https://clang.llvm.org/extra/clang-tidy/checks/readability-else-after-return.html). This project hasn't formally adopted that style though so it's your call.